packages/ovm/src/contracts/ExecutionManager.sol

```diff
@@ -5,26 +5,30 @@ pragma experimental ABIEncoderV2;
 import {DataTypes as dt} from "./DataTypes.sol";
 import {FullStateManager} from "./FullStateManager.sol";
 import {ContractAddressGenerator} from "./ContractAddressGenerator.sol";
-import {CreatorContract} from "./CreatorContract.sol";
 import {PurityChecker} from "./PurityChecker.sol";
 import {RLPEncode} from "./RLPEncode.sol";
-import {L2ToL1MessagePasser} from "./L2ToL1MessagePasser.sol";
+import {L2ToL1MessagePasser} from "./precompiles/L2ToL1MessagePasser.sol";
+import {L1MessageSender} from "./precompiles/L1MessageSender.sol";
 
 /**
  * @title ExecutionManager
  * @notice The execution manager ensures that the execution of each transaction is sandboxed in a distinct environment as defined
  *         by the supplied backend. Only state / contracts from that backend will be accessed.
  */
 contract ExecutionManager is FullStateManager {
+    address ZERO_ADDRESS = 0x0000000000000000000000000000000000000000;
+
+    // expected queue origin for calls from L1
+    uint constant L1_QUEUE_ORIGIN = 1;
+
     // bitwise right shift 28 * 8 bits so the 4 method ID bytes are in the right-most bytes
     bytes32 constant ovmCallMethodId = keccak256("ovmCALL()") >> 224;
     bytes32 constant ovmCreateMethodId = keccak256("ovmCREATE()") >> 224;
     bytes32 constant executeCallMethodId = keccak256("executeCall()") >> 224;
 
-    // creator contract address
-    address constant creatorContractAddress = 0x0000000000000000000000000000000000000000;
-    address ZERO_ADDRESS = 0x0000000000000000000000000000000000000000;
+    // Precompile addresses
     address constant l2ToL1MessagePasserOvmAddress = 0x4200000000000000000000000000000000000000;
+    address constant l1MsgSenderAddress = 0x4200000000000000000000000000000000000001;
```

Comment:
Ah no reason. These are just the reserved addresses for precompiles. We used these values because Ethereum's precompiles are located at addresses `0x0...0`, `0x0...1`, `0x0...2`, etc.

 Definitely down to make them a bit distinct though